shared/params/config.go

```diff
@@ -119,9 +120,12 @@ type BeaconChainConfig struct {
 }
 
 var beaconConfig = MainnetConfig()
+var lock sync.RWMutex
 
 // BeaconConfig retrieves beacon chain config.
 func BeaconConfig() *BeaconChainConfig {
+	lock.RLock()
```

Comment:
whats with the locks here ?

Reply:
please review and merge #8114 first, this PR has been moved to draft